lib/WP_Auth0_LoginManager.php

```diff
@@ -502,71 +504,79 @@ private function do_login( $user, $userinfo, $is_new, $id_token, $access_token )
 	/**
 	 * Complete the logout process based on settings.
 	 * Hooked to `wp_logout` action.
+	 * IMPORTANT: Internal callback use only, do not call this function directly!
 	 *
 	 * @see WP_Auth0_LoginManager::init()
+	 *
+	 * @link https://codex.wordpress.org/Plugin_API/Action_Reference/wp_logout
 	 */
 	public function logout() {
-		$this->end_session();
-
-		$sso        = $this->a0_options->get( 'sso' );
-		$slo        = $this->a0_options->get( 'singlelogout' );
-		$client_id  = $this->a0_options->get( 'client_id' );
-		$auto_login = absint( $this->a0_options->get( 'auto_login' ) );
-
-		if ( $slo && isset( $_REQUEST['SLO'] ) && isset( $_REQUEST['redirect_to'] ) ) {
-			wp_redirect( $_REQUEST['redirect_to'] );
+		$is_sso        = (bool) $this->a0_options->get( 'sso' );
+		$is_slo        = (bool) $this->a0_options->get( 'singlelogout' );
+		$is_auto_login = (bool) $this->a0_options->get( 'auto_login' );
+
+		// Redirected here after checkSession in the footer (templates/auth0-singlelogout-handler.php).
+		if ( $is_slo && isset( $_REQUEST['SLO'] ) ) {
+			if ( ! empty( $_REQUEST['redirect_to'] ) && filter_var( $_REQUEST['redirect_to'], FILTER_VALIDATE_URL ) ) {
+				$redirect_url = $_REQUEST['redirect_to'];
+			} else {
+				$redirect_url = home_url();
+			}
+			wp_redirect( $redirect_url );
 			exit;
```

Comment:
at this line `$is_sso` and `$is_auto_login` were not used and the function is already returning. Would it make sense to define those variables right above the line that uses it? Would that help this code perform better?

Reply:
I think that's less clear and `$this->a0_options->get()` is just returning an array value so I don't think optimization there is helpful.